### PR TITLE
V13 API 6.2 implementation

### PR DESCRIPTION
**Pull request by Poolitzer**
### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs and all suitable `__all__` s

### If the PR contains API changes (otherwise, you can delete this passage)

* New classes:
    - [ ] Added `self._id_attrs` and corresponding documentation
    - [ ] `__init__` accepts `**_kwargs`
    
* Added new shortcuts:
    - [ ] In `Chat` & `User` for all methods that accept `chat/user_id`
    - [ ] In `Message` for all methods that accept `chat_id` and `message_id`
    - [ ] For new `Message` shortcuts: Added `quote` argument if methods accepts `reply_to_message_id`
    - [ ] In `CallbackQuery` for all methods that accept either `chat_id` and `message_id` or `inline_message_id`
    
* If relevant:

    - [ ] Added new constants at `telegram.constants` and shortcuts to them as class variables
    - [ ] Link new and existing constants in docstrings instead of hard coded number and strings
    - [ ] Add new message types to `Message.effective_attachment`
    - [ ] Added new handlers for new update types
      - [ ] Add the handlers to the warning loop in the `ConversationHandler`
    - [ ] Added new filters for new message (sub)types
    - [ ] Added or updated documentation for the changed class(es) and/or method(s)
    - [ ] Added or updated `bot_methods.rst`
    - [ ] Updated the Bot API version number in all places: `README.rst` and `README_RAW.rst` (including the badge), as well as `telegram.constants.BOT_API_VERSION`
    - [ ] Added logic for arbitrary callback data in `tg.ext.Bot` for new methods that either accept a `reply_markup` in some form or have a return type that is/contains `telegram.Message`

### CODE REVIEW

File: telegram/files/sticker.py [hunk 1]
```diff
@@ -60,10 +65,14 @@ class Sticker(TelegramObject):
             position where the mask should be placed.
         file_size (:obj:`int`, optional): File size.
         bot (:class:`telegram.Bot`, optional): The Bot to use for instance methods.
-        premium_animation (:class:`telegram.File`, optional): Premium animation for the sticker,
-            if the sticker is premium.
+        premium_animation (:class:`telegram.File`, optional): For premium regular stickers,
+            premium animation for the sticker.
+
+            .. versionadded:: 13.14
```
Comment (Bibo-Joshi) [inline telegram/files/sticker.py:71]:
```suggestion
            .. versionadded:: 13.13
```

File: telegram/files/sticker.py [hunk 2]
```diff
@@ -77,17 +86,26 @@ class Sticker(TelegramObject):
...
         set_name (:obj:`str`): Optional. Name of the sticker set to which the sticker belongs.
         mask_position (:class:`telegram.MaskPosition`): Optional. For mask stickers, the position
             where the mask should be placed.
         file_size (:obj:`int`): Optional. File size.
-        premium_animation (:class:`telegram.File`): Optional. Premium animation for the sticker,
-            if the sticker is premium.
+        premium_animation (:class:`telegram.File`): Optional. For premium regular stickers,
+            premium animation for the sticker.
+
+            .. versionadded:: 13.14
```
Comment (Bibo-Joshi) [inline telegram/files/sticker.py:104]:
```suggestion
            .. versionadded:: 13.13
```

File: telegram/files/sticker.py [hunk 3]
```diff
@@ -178,6 +202,13 @@ def get_file(
         """
         return self.bot.get_file(file_id=self.file_id, timeout=timeout, api_kwargs=api_kwargs)
 
+    REGULAR: ClassVar[str] = constants.STICKER_REGULAR
+    """:const:`telegram.constants.STICKER_REGULAR`"""
+    MASK: ClassVar[str] = constants.STICKER_MASK
+    """:const:`telegram.constants.STICKER_MASK`"""
+    CUSTOM_EMOJI: ClassVar[str] = constants.STICKER_CUSTOM_EMOJI
+    """:const:`telegram.constants.STICKER_CUSTOM_EMOJI`"""
```
Comment (Bibo-Joshi) [inline telegram/files/sticker.py:210]: versionadded

File: telegram/files/sticker.py [hunk 4]
```diff
@@ -190,6 +221,10 @@ class StickerSet(TelegramObject):
         arguments had to be changed. Use keyword arguments to make sure that the arguments are
         passed correctly.
 
+    .. versionchanged:: 13.14:
+        The parameter ``contains_masks`` has been depreciated.
+        Use ``sticker_type`` instead.
```
Comment (Bibo-Joshi) [inline telegram/files/sticker.py:226]: Maybe add something like "As of Bot API 6.2, the parameter …"? also in the other locations where this is relevant

File: telegram/message.py [hunk 5]
```diff
@@ -2752,6 +2754,9 @@ def text_html(self) -> str:
         Use this if you want to retrieve the message text with the entities formatted as HTML in
         the same way the original message was formatted.
 
+        Note:
+            |custom_emoji_formatting_note|
```
Comment (Bibo-Joshi) [inline telegram/message.py:2758]: Just curious: Can the `Note:` be part of the replace thingy?

File: telegram/message.py [hunk 6]
```diff
@@ -2986,6 +3000,8 @@ def text_markdown(self) -> str:
             :attr:`telegram.ParseMode.MARKDOWN` is is a legacy mode, retained by Telegram for
             backward compatibility. You should use :meth:`text_markdown_v2` instead.
 
+            |custom_emoji_formatting_note|
+
```
Comment (Bibo-Joshi) [inline telegram/message.py:3004]: Note:

File: telegram/message.py [hunk 7]
```diff
@@ -3024,6 +3043,8 @@ def text_markdown_urled(self) -> str:
             :attr:`telegram.ParseMode.MARKDOWN` is is a legacy mode, retained by Telegram for
             backward compatibility. You should use :meth:`text_markdown_v2_urled` instead.
 
+            |custom_emoji_formatting_note|
```
Comment (Bibo-Joshi) [inline telegram/message.py:3046]: Note:

File: telegram/message.py [hunk 8]
```diff
@@ -3062,6 +3086,8 @@ def caption_markdown(self) -> str:
             :attr:`telegram.ParseMode.MARKDOWN` is is a legacy mode, retained by Telegram for
             backward compatibility. You should use :meth:`caption_markdown_v2` instead.
 
+            |custom_emoji_formatting_note|
+
```
Comment (Bibo-Joshi) [inline telegram/message.py:3090]: Note:

File: telegram/message.py [hunk 9]
```diff
@@ -3102,6 +3131,8 @@ def caption_markdown_urled(self) -> str:
             :attr:`telegram.ParseMode.MARKDOWN` is is a legacy mode, retained by Telegram for
             backward compatibility. You should use :meth:`caption_markdown_v2_urled` instead.
 
+            |custom_emoji_formatting_note|
```
Comment (Bibo-Joshi) [inline telegram/message.py:3134]: .

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the fast PR!

* `test_official` is complaining that `StickerSet.contains_masks` is still present. Can you check how we handled such cases in the past?
* DeepSource is complaining about two places where we could use f-strings. Feel free to improve, or just add a `# skipcq: …`

**Review by neilarchie06 — APPROVED** [review]

**Review by neilarchie06 — APPROVED** [review]

File: telegram/message.py (hunk 5 above)
Comment (Poolitzer) [reply telegram/message.py:2758]: I havent found a way, it breaks the indention I think

File: telegram/message.py (hunk 6 above)
Comment (Poolitzer) [reply telegram/message.py:3004]: no its in the upper block

File: telegram/message.py (hunk 7 above)
Comment (Poolitzer) [reply telegram/message.py:3046]: same

File: telegram/message.py (hunk 8 above)
Comment (Poolitzer) [reply telegram/message.py:3090]: same

File: telegram/message.py (hunk 9 above)
Comment (Poolitzer) [reply telegram/message.py:3134]: -

File: telegram/message.py (hunk 5 above)
Comment (Bibo-Joshi) [reply telegram/message.py:2758]: sad :(

File: telegram/message.py (hunk 6 above)
Comment (Bibo-Joshi) [reply telegram/message.py:3004]: ah sorry, hadn't expanded the diff … but can we make it an itemized list, then?

Comment (Poolitzer) [reply telegram/message.py:3004]: You just want me to work more grrr.

Will do.

**Comment by Bibo-Joshi** [comment]
Thanks for the updates. I think I'm happy with the changes, then. I vote to ignore DS on this for the reasons mentioned in https://discuss.deepsource.io/t/improve-granular-setting-possibility-to-pin-a-deepsource-version/472.

**Pull request merged**